src/crimson/osd/shard_services.h

```diff
@@ -90,8 +91,21 @@ class ShardServices : public md_config_obs_t {
 
   template <typename T, typename... Args>
   auto start_operation(Args&&... args) {
-    auto op = registry.create_operation<T>(std::forward<Args>(args)...);
-    return std::make_pair(op, op->start());
+    try {
+      operation_gate.enter();
+      auto op = registry.create_operation<T>(std::forward<Args>(args)...);
+      auto p = std::make_pair(op, op->start());
+      operation_gate.leave();
```

Comment:
nit, could you use?
```c++
return with_gate(operation_gate, [&] {
  auto op = registry.create_operation<T>(std::forward<Args>(args)...);
  return std::make_pair(op, op->start());
}
```

Reply:
@tchaikov It seems that this may turn the original synchorous operation creation into an asynchorous one, which would involve a series of other modifications and may lead to potential operation order problem. Maybe we should postpone this modification, how do you like it?:-)

Reply:
@tchaikov I shouldn't use Gate in the first place, I just wanted to prevent operation creation when shutting down OSDs.